### ISSUE TITLE
Error importing tensorflow.  Unless you are using bazel, you should not try to import tensorflow from its source directory

### ISSUE THREAD

**Issue by ushnish** [tensorflow/tensorflow#1402]
These are the sequence of steps i followed to install the distributed version of TF

> > git clone --recurse-submodules https://github.com/tensorflow/tensorflow
> > ./configure (default with GPU)
> > bazel build -c opt --config=cuda //tensorflow/cc:tutorials_example_trainer
> > bazel-bin/tensorflow/cc/tutorials_example_trainer --use_gpu (at this point saw the expected output and GPU being used)
> >  bazel build -c opt --config=cuda --define=use_fast_cpp_protos=true //tensorflow/tools/pip_package:build_pip_package
> > bazel-bin/tensorflow/tools/pip_package/build_pip_package /tmp/tensorflow_pkg
> > sudo pip install /tmp/tensorflow_pkg/tensorflow-0.7.1-py2-none-any.whl 
> > Processing /tmp/tensorflow_pkg/tensorflow-0.7.1-py2-none-any.whl
> > Installing collected packages: tensorflow
> > pip show tensorflow
> > Metadata-Version: 2.0
> > Name: tensorflow
> > Version: 0.7.1
> > Summary: TensorFlow helps the tensors flow
> > Home-page: http://tensorflow.org/
> > Author: Google Inc.
> > Author-email: opensource@google.com
> > License: Apache 2.0
> > Location: /usr/local/lib/python2.7/dist-packages
> > Requires: six, protobuf, wheel, numpy
> > Classifiers:
> >   Development Status :: 4 - Beta
> >   Intended Audience :: Developers
> >   Intended Audience :: Education
> >   Intended Audience :: Science/Research
> >   License :: OSI Approved :: Apache Software License
> >   Programming Language :: Python :: 2.7
> >   Topic :: Scientific/Engineering :: Mathematics
> >   Topic :: Software Development :: Libraries :: Python Modules
> >   Topic :: Software Development :: Libraries
> > Entry-points:
> >   [console_scripts]
> >   tensorboard = tensorflow.tensorboard.tensorboard:main

Now for the error

> > python
> > Python 2.7.9 (default, Apr  2 2015, 15:33:21) 
> > [GCC 4.9.2] on linux2
> > Type "help", "copyright", "credits" or "license" for more information.
> > 
> > > import tensorflow
> > > Traceback (most recent call last):
> > >   File "<stdin>", line 1, in <module>
> > >   File "/usr/local/lib/python2.7/dist-packages/tensorflow/**init**.py", line 23, in <module>
> > >     from tensorflow.python import *
> > >   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/**init**.py", line 41, in <module>
> > >     raise ImportError(msg)
> > > ImportError: Traceback (most recent call last):
> > >   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/**init**.py", line 35, in <module>
> > >     from tensorflow.core.framework.graph_pb2 import *
> > >   File "/usr/local/lib/python2.7/dist-packages/tensorflow/core/framework/graph_pb2.py", line 6, in <module>
> > >     from google.protobuf import descriptor as _descriptor
> > > ImportError: No module named protobuf

Error importing tensorflow.  Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter
from there.

Can someone help me understand what might cause this? Thank you.

**Comment by shiyemin** [comment]
You should start your python outside tensorflow directory. Because python will import tensorflow from current directory.

**Comment by raingo** [comment]
I had the same errors as you did. I solved this by reinstall protobuf, by `pip install -U protobuf==3.0.0b2`

**Comment by ushnish** [comment]
@shiyemin yes of course I started the interpreter outside the tensorflow directory

@raingo I tried that but in python interpreter I still get same error. Do you have to do a bazel build for tensorflow by pass in this protobuf library into the build command?

**Comment by raingo** [comment]
Yes. I built from source, but I didn't do anything special for bazel. I
needed to upgrade protobuf manually because I install the wheel by

pip install wheel --no-deps -U

**Comment by ushnish** [comment]
@raingo did you do **git clone --recurse-submodules https://github.com/tensorflow/tensorflow** or clone without the submodules since you are installing protobuf separately

**Comment by Dringite** [comment]
Uninstall protobuf before building (or dl whl), everything works fine after that. Had the same issue.

sudo pip uninstall protobuf

**Comment by ushnish** [comment]
@Dringite do you mean git clone with the --recurse-submodules options and then uninstall protobuf? But that'll cause the bazel build to fail?

**Comment by ushnish** [comment]
Problem solved @Dringite you were right, there was no need to have protobuf installed at all. The steps are

>> git clone --recurse-submodules https://github.com/tensorflow/tensorflow 
>> sudo pip uninstall protobuf
>> cd tensorflow/
>> ./configure 
>> bazel build -c opt --config=cuda **--define=use_fast_cpp_protos=true** //tensorflow/tools/pip_package:build_pip_package
>> bazel-bin/tensorflow/tools/pip_package/build_pip_package /tmp/tenso
rflow_pkg
>> sudo pip install /tmp/tensorflow_pkg/tensorflow-0.7.1-py2-none-any.whl 

Notice that i added --define=use_fast_cpp_protos=true in the build command so it might have actually helped.

**Comment by Channndiii** [comment]
@ushnish I have followed your way to install tensorflow-0.7.1. In the future if I want to upgrade the version of tensorflow, is it necessary for me to follow your way to install new version? Or is there any other convenient ways?

**Comment by hhchen1105** [comment]
Tensorflow 0.9.0 also suffers from the same issue.   The answer from @Dringite can solve the issue.  (i.e., pip uninstall protobuf and tensorflow, then pip install tensorflow)

**Comment by Ethanator** [comment]
Tensorflow 0.10.0 also suffers from the same issue. The answer from @raingo can solve the issue. (i.e., `pip uninstall protobuf`, then `pip install wheel --no-deps -U`)

**Comment by 984958198** [comment]
According to the above method Still not solve
ubuntu 64
tensorflow-0.12.0rc0-cp27-cp27mu-manylinux1_x86_64.whl

`Python 2.7.3 (default, Jun 22 2015, 19:33:41) 
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/root/tg2env/local/lib/python2.7/site-packages/tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "/root/tg2env/local/lib/python2.7/site-packages/tensorflow/python/__init__.py", line 60, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "/root/tg2env/local/lib/python2.7/site-packages/tensorflow/python/__init__.py", line 49, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "/root/tg2env/local/lib/python2.7/site-packages/tensorflow/python/pywrap_tensorflow.py", line 28, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "/root/tg2env/local/lib/python2.7/site-packages/tensorflow/python/pywrap_tensorflow.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow', fp, pathname, description)
ImportError: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.16' not found (required by /root/tg2env/local/lib/python2.7/site-packages/tensorflow/python/_pywrap_tensorflow.so)


Error importing tensorflow.  Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter
from there.
`

**Comment by baregawi** [comment]
I'm running Ubuntu 14.04.5 LTS with `tensorflow==0.12.0rc0` (most recent in PyPy) and keras==1.1.0.

A quick hack to get you by:
```
sudo vim /usr/local/lib/python2.7/dist-packages/tensorflow/python/__init__.py
```
and replace `from tensorflow.python import pywrap_tensorflow` with `import pywrap_tensorflow` around line 49 in the try portion of the try/catch.

**Comment by NewRem** [comment]
I got a similar error. It was solved when I run python from the parent directory of tensorflow. But I'm curious if I can import tensorflow from another directory? Because when I do, it shows this error.
"Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter
from there."

**Comment by martinwicke** [comment]
@baregawi Is this a new behavior in 0.12? 

@NewRem how did you install TensorFlow?

@yifeif It obviously looks fine in our CI, any idea?

**Comment by yifeif** [comment]
Looks like it is related to this line when running tensorflow from source dir, https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/__init__.py#L49

This should be resolved by not running from the source dir.
@NewRem also curious how you installed tensorflow. You should be able to import tensorflow from a non-repo-source directory.

**Comment by 984958198** [comment]
Thank you all, I solved the problem by upgrading the ubuntu

**Comment by NewRem** [comment]
@martinwicke @yifeif I used pip installation. To be exact, I used these three commands:

$ sudo easy_install pip
$ sudo easy_install --upgrade six

#the CPU version of the binaries to Pypi
$ sudo pip install --ignore-installed tensorflow

**Comment by baregawi** [comment]
@martinwicke I just `pip install -U tensorflow` after not having updated since September. But I remember having to do a similar hack in September: https://github.com/tensorflow/tensorflow/issues/4616

**Comment by martinwicke** [comment]
@baregawi #4616 is different, we are removing spurious symbols from our packages, so a lot of these direct imports will stop working. 

This issue is different, and in the past, this has pointed to tensorflow now being properly installed. It's hard to debug remotely though. @ushnish how many pythons do you have installed and which one is tensorflow installed in?

Can you install tensorflow in a virtualenv? If that works, then there's some problem with conflicting versions, either python/ipython or more than one tensorflow installation.

**Comment by R0rs12ach** [comment]
@martinwicke I installed **tensorflow_gpu-0.12rc1-py2-none-any.whl** step by step via the instruction of [official site](https://www.tensorflow.org/versions/master/get_started/os_setup.html#protobuf-library-related-issues). 

execute this command: `sudo pip install --ignore-installed --upgrade https://storage.googleapis.com/tensorflow/mac/gpu/tensorflow_gpu-0.12.0rc1-py2-none-any.whl`

```
Collecting tensorflow-gpu==0.12.0rc1 from https://storage.googleapis.com/tensorflow/mac/gpu/tensorflow_gpu-0.12.0rc1-py2-none-any.whl
  Downloading https://storage.googleapis.com/tensorflow/mac/gpu/tensorflow_gpu-0.12.0rc1-py2-none-any.whl (83.6MB)
    100% |████████████████████████████████| 83.6MB 7.5kB/s
Collecting mock>=2.0.0 (from tensorflow-gpu==0.12.0rc1)
  Downloading mock-2.0.0-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 33kB/s
Collecting numpy>=1.11.0 (from tensorflow-gpu==0.12.0rc1)
  Downloading numpy-1.11.2-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (3.9MB)
    100% |████████████████████████████████| 3.9MB 16kB/s
Collecting protobuf==3.1.0 (from tensorflow-gpu==0.12.0rc1)
  Downloading protobuf-3.1.0-py2.py3-none-any.whl (339kB)
    100% |████████████████████████████████| 348kB 16kB/s
Collecting wheel (from tensorflow-gpu==0.12.0rc1)
  Downloading wheel-0.29.0-py2.py3-none-any.whl (66kB)
    100% |████████████████████████████████| 71kB 19kB/s
Collecting six>=1.10.0 (from tensorflow-gpu==0.12.0rc1)
  Downloading six-1.10.0-py2.py3-none-any.whl
Collecting funcsigs>=1; python_version < "3.3" (from mock>=2.0.0->tensorflow-gpu==0.12.0rc1)
  Downloading funcsigs-1.0.2-py2.py3-none-any.whl
Collecting pbr>=0.11 (from mock>=2.0.0->tensorflow-gpu==0.12.0rc1)
  Downloading pbr-1.10.0-py2.py3-none-any.whl (96kB)
    100% |████████████████████████████████| 102kB 28kB/s
Collecting setuptools (from protobuf==3.1.0->tensorflow-gpu==0.12.0rc1)
  Downloading setuptools-32.0.0-py2.py3-none-any.whl (477kB)
    100% |████████████████████████████████| 481kB 30kB/s
Installing collected packages: six, funcsigs, pbr, mock, numpy, setuptools, protobuf, wheel, tensorflow-gpu
Successfully installed funcsigs-1.0.2 mock-2.0.0 numpy-1.11.2 pbr-1.10.0 protobuf-3.1.0 setuptools-32.0.0 six-1.10.0 tensorflow-gpu-0.12.0rc1 wheel-0.29.0
```
But unfortunately, I got the same error as following output:
```
>>> import tensorflow
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "/Library/Python/2.7/site-packages/tensorflow/python/__init__.py", line 60, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/tensorflow/python/__init__.py", line 49, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "/Library/Python/2.7/site-packages/tensorflow/python/pywrap_tensorflow.py", line 28, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "/Library/Python/2.7/site-packages/tensorflow/python/pywrap_tensorflow.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow', fp, pathname, description)
ImportError: dlopen(/Library/Python/2.7/site-packages/tensorflow/python/_pywrap_tensorflow.so, 10): no suitable image found.  Did find:
	/Library/Python/2.7/site-packages/tensorflow/python/_pywrap_tensorflow.so: mach-o, but wrong architecture


Error importing tensorflow.  Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter
from there.
>>> exit()
```
The version of mac system: `OS X EI Captain`
And I had try each ways above this issues list, but nothing change.

Thank u 4 any response.

**Comment by martinwicke** [comment]
@R0rs12ach:
The hint is in this line: 
Did find: /Library/Python/2.7/site-packages/tensorflow/python/_pywrap_tensorflow.so: mach-o, but wrong architecture

Are you installing this on PPC? If so, you'll have to install from source, our binary packages only support x86_64.

**Comment by MeilingShi** [comment]
@R0rs12ach: i had the same problem using mac OS X Yosemite.
i have tried everything written here but it didn't help.
At last i went back to the homepage of tensor flow and tried other installation methods. After that i have followed 'Anaconda installation' -> 'Using conda' i finally could run it properly. Not sure which action i have taken helped.

So far i have tried:
```
$ sudo easy_install pip
$ sudo easy_install --upgrade six

$ sudo pip install --upgrade $TF_BINARY_URL
```


```
pip uninstall protobuf
pip install wheel --no-deps -U)

git clone --recurse-submodules https://github.com/tensorflow/tensorflow
sudo pip uninstall protobuf
cd tensorflow/
./configure
bazel build -c opt --config=cuda --define=use_fast_cpp_protos=true //tensorflow/tools/pip_package:build_pip_package
bazel-bin/tensorflow/tools/pip_package/build_pip_package /tmp/tenso
rflow_pkg
sudo pip install /tmp/tensorflow_pkg/tensorflow-0.7.1-py2-none-any.whl

```

and as said, instead of install with pip i have made it with Anaconda installation from trensorflow homepage: https://www.tensorflow.org/get_started/os_setup#anaconda_installation

Hope it will help you.

**Comment by R0rs12ach** [comment]
@martinwicke I think my computer's arch is :

```
Darwin MacBook-Pro.local 15.6.0 Darwin Kernel Version 15.6.0: Thu Jun 23 18:25:34 PDT 2016; root:xnu-3248.60.10~1/RELEASE_X86_64 x86_64
```
the fact is that my computer just x86_64

@MeilingShi : Thank u 4 ur advice, I will try it and paste the result

**Comment by R0rs12ach** [comment]
@MeilingShi I have tried what you said above, but nothing change. In fact, the cuda sdk do not support intel graphic 4000. So, i give up to install tensorflow_gpu_version on my mac which bought at 2013.

**Comment by martinwicke** [comment]
@R0rs12ach yes, CUDA only works with NVIDIA graphics cards. 

It's still confusing me that it would complain about the library architecture. 

Others, I would like to understand this issue, but this thread has become too noisy. Whoever still has this problem, can you open a new issue with your system config according to the template?

**Comment by NuovoVesuvio** [comment]
I follow the official pip windows installation and get same error as others.

I also tried uninstalling protobuf etc beforehand to no avail. I'm running python 3.5.2 and Anaconda. 

Is everyone having this problem on Windows?

`C:\Users\Rafie>python
Python 3.5.2 |Anaconda 4.2.0 (64-bit)| (default, Jul  5 2016, 11:41:13) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:119] Couldn't open CUDA library cublas64_80.dll
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_blas.cc:2294] Unable to load cuBLAS DSO.
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:119] Couldn't open CUDA library cudnn64_5.dll
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_dnn.cc:3459] Unable to load cuDNN DSO
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:119] Couldn't open CUDA library cufft64_80.dll
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_fft.cc:344] Unable to load cuFFT DSO.
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:119] Couldn't open CUDA library nvcuda.dll
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_diagnostics.cc:165] hostname: °9l├ 
Traceback (most recent call last):
  File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Program Files\Anaconda3\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 906, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: A dynamic link library (DLL) initialization routine failed.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\__init__.py", line 54, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow')
  File "C:\Program Files\Anaconda3\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\__init__.py", line 60, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Program Files\Anaconda3\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 906, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: A dynamic link library (DLL) initialization routine failed.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\__init__.py", line 54, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow')
  File "C:\Program Files\Anaconda3\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow'


Error importing tensorflow.  Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter
from there.`

**Comment by pattrickrice** [comment]
Same issue as @NuovoVesuvio

**Comment by anddelu** [comment]
my environment:
centos6.5 x86_64
python 2.7

I use pip install tensorflow version1.2 succuessfully. However I enter python and import tensorflow, I met the error like above. I try above solution, but it still failed.
ImportError: /lib64/libc.so.6: version `GLIBC_2.16' not found (required by /data/apps/anaconda/lib/python2.7/site-packages/tensorflow/python/_pywrap_tensorflow.so)

Error importing tensorflow.  Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter

Anyone has ideas for it? Thanks very much!

**Comment by martinwicke** [comment]
The binaries we provide do not work on CentOS 6.5, because the system libraries are too old. You will have to build from source.

**Comment by i-xying** [comment]
I really have the same error. but when i update my tensorflow, it's resolved.
sudo yum install python-pip python-dev
export TF_BINARY_URL=https://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-0.12.1-cp27-none-linux_x86_64.whl
sudo pip install --upgrade $TF_BINARY_URL

**Comment by wiso** [comment]
I have the same problem on fedora 25. I tried all the proposed solution. It works for tensorflow, but not for tensorflow-gpu

**Comment by rodrigomfw** [comment]
Same problem: "Error importing tensorflow. Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter"

Tried building from source, installing through pip. All the same. I will just move to theano

**Comment by martinwicke** [comment]
@rodrigomfw can you paste the error before that message? For some people on this thread it's that they have very old systems (CentOS 6.x) which don't support the pre-built binaries. For you it's probably something else since it only affects GPU. 

@vrv, I think we should rephrase this error. Most of the time (100% of the time in this thread), this is a failure to import unrelated to the working directory.

**Comment by vrv** [comment]
Yup, something changed and we now raise ImportError much more often, so the error message is unhelpful.

**Comment by MumMumMum** [comment]
I just uninstalled tensorflow and tensorflow-gpu both and then I just installed tensorflow-gpu... works perfectly fine for me

**Comment by shankyjari** [comment]
Facing this error when installing tensorflow on windows 8 with python 3.5 and importing it from IDLE. I think it is the same issue.

>>> import tensorflow
Traceback (most recent call last):
  File "C:\python_35_2_64\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\python_35_2_64\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 906, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: A dynamic link library (DLL) initialization routine failed.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\python_35_2_64\lib\site-packages\tensorflow\python\__init__.py", line 54, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\python_35_2_64\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\python_35_2_64\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow')
  File "C:\python_35_2_64\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<pyshell#0>", line 1, in <module>
    import tensorflow
  File "C:\python_35_2_64\lib\site-packages\tensorflow\__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "C:\python_35_2_64\lib\site-packages\tensorflow\python\__init__.py", line 60, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "C:\python_35_2_64\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\python_35_2_64\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 906, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: A dynamic link library (DLL) initialization routine failed.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\python_35_2_64\lib\site-packages\tensorflow\python\__init__.py", line 54, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\python_35_2_64\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\python_35_2_64\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow')
  File "C:\python_35_2_64\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow'


Error importing tensorflow.  Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter
from there.

**Comment by yemengsky** [comment]
pip install --upgrade https://storage.googleapis.com/tensorflow/mac/cpu/protobuf-3.1.0-cp27-none-macosx_10_11_x86_64.whl

**Comment by gistya** [comment]
I did everything above but I always get this: 

> Using TensorFlow backend.
> RuntimeError: module compiled against API version 0xa but this version of numpy is 0x9
> Traceback (most recent call last):
>   File "run.py", line 1, in <module>
>     import lstm
>   File "/Users/jjj/Documents/dev/LSTM/lstm.py", line 5, in <module>
>     from keras.layers.core import Dense, Activation, Dropout
>   File "/Library/Python/2.7/site-packages/Keras-1.2.0-py2.7.egg/keras/__init__.py", line 2, in <module>
>     from . import backend
>   File "/Library/Python/2.7/site-packages/Keras-1.2.0-py2.7.egg/keras/backend/__init__.py", line 67, in <module>
>     from .tensorflow_backend import *
>   File "/Library/Python/2.7/site-packages/Keras-1.2.0-py2.7.egg/keras/backend/tensorflow_backend.py", line 1, in <module>
>     import tensorflow as tf
>   File "/Library/Python/2.7/site-packages/tensorflow/__init__.py", line 24, in <module>
>     from tensorflow.python import *
>   File "/Library/Python/2.7/site-packages/tensorflow/python/__init__.py", line 60, in <module>
>     raise ImportError(msg)
> ImportError: Traceback (most recent call last):
>   File "/Library/Python/2.7/site-packages/tensorflow/python/__init__.py", line 49, in <module>
>     from tensorflow.python import pywrap_tensorflow
>   File "/Library/Python/2.7/site-packages/tensorflow/python/pywrap_tensorflow.py", line 28, in <module>
>     _pywrap_tensorflow = swig_import_helper()
>   File "/Library/Python/2.7/site-packages/tensorflow/python/pywrap_tensorflow.py", line 24, in swig_import_helper
>     _mod = imp.load_module('_pywrap_tensorflow', fp, pathname, description)
> ImportError: numpy.core.multiarray failed to import
> 
> 
> Error importing tensorflow.  Unless you are using bazel,
> you should not try to import tensorflow from its source directory;
> please exit the tensorflow source tree, and relaunch your python interpreter
> from there.

**Comment by yemengsky** [comment]
@gistya https://www.tensorflow.org/get_started/os_setup

**Comment by atabakd** [comment]
I got over the problem by reinstalling cuda 8.0 and cudnn 5.1

**Comment by luckyuho** [comment]
Maybe this can help you when you got this "Error importing tensorflow. Unless you are using bazel, you should not try to import tensorflow from its source directory"  https://github.com/tensorflow/tensorflow/issues/559

**Comment by luckyuho** [comment]
try this one
https://github.com/tensorflow/tensorflow/issues/559
from author se7en007 commented on 4 Jan 2016
import numpy as np
np.path
['/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy']
exit()
sudo rm -rf /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy
...

**Comment by clarklight** [comment]
If anyone had the same problem as mine, i managed to fix it. My error was
 **ImportError: numpy.core.multiarray failed to import
Error importing tensorflow. Unless yo......**

Because i am running on a Mac, there are 2 sets of numpy thats running at the same time, as the Mac came with Python installed, so even if you are updating the numpy, it wasnt updating the one thats needed for Tensorflow. I believe i ran sudo pip install numpy -U --upgrade or something like that so all versions of numpy would be upgraded within my Mac. Then it worked after that.

**Comment by brando90** [comment]
What does it mean to be in the "source directory". I am also having the same issue however, I have no idea what the source directory is.

**Comment by jhli973** [comment]
seems no solution yet. I am installing gpu version on windows 10. The installation seems ok at least no error or warning. I have 

pip install --ignore-installed --upgrade C:\Users\dbsnail\Downloads\tensorflow_gpu-1.0.0rc2-cp35-cp35m-win_amd64.whl


>>> import tensorflow
Traceback (most recent call last):
  File "C:\Anaconda2\envs\py35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Anaconda2\envs\py35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 906, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Anaconda2\envs\py35\lib\site-packages\tensorflow\python\__init__.py", line 66, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Anaconda2\envs\py35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\Anaconda2\envs\py35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow')
  File "C:\Anaconda2\envs\py35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Anaconda2\envs\py35\lib\site-packages\tensorflow\__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "C:\Anaconda2\envs\py35\lib\site-packages\tensorflow\python\__init__.py", line 72, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "C:\Anaconda2\envs\py35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Anaconda2\envs\py35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 906, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Anaconda2\envs\py35\lib\site-packages\tensorflow\python\__init__.py", line 66, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Anaconda2\envs\py35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\Anaconda2\envs\py35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow')
  File "C:\Anaconda2\envs\py35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow'


Failed to load the native TensorFlow runtime.

See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/get_started/os_setup.md#import_error

for some common reasons and solutions.  Include the entire stack trace
above this error message when asking for help.

**Comment by jhli973** [comment]
I got the solution, install cuDNN v5.1 for CUDA 8.0
You need to register as a NVIDIA accelerated computing developer program, then from here (https://developer.nvidia.com/rdp/cudnn-download) download and unzip it.
Copy all three subfolders and past it under CUDA folder, in my case is
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA

Then , you should be able to load tensorflow without any error

NOTE, I use python 3.5

**Comment by mzmmoazam** [comment]
I first installed wheel, then the tensorflow was installed on my machine(using windows 10).

Now, it is not getting imported here is the error :
  
Using TensorFlow backend.
Traceback (most recent call last):
  File "C:\Users\mzm\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    fp, pathname, description = imp.find_module('_pywrap_tensorflow', [dirname(__file__)])
  File "C:\Users\mzm\Anaconda3\lib\imp.py", line 296, in find_module
    raise ImportError(_ERR_MSG.format(name), name=name)
ImportError: No module named '_pywrap_tensorflow'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\mzm\Anaconda3\lib\site-packages\tensorflow\python\__init__.py", line 54, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Users\mzm\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 29, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\Users\mzm\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in swig_import_helper
    import _pywrap_tensorflow
ModuleNotFoundError: No module named '_pywrap_tensorflow'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:/Users/mzm/PycharmProjects/mzm/FAQ bot/rough.py", line 36, in <module>
    from keras.models import Sequential, Model
  File "C:\Users\mzm\Anaconda3\lib\site-packages\keras\__init__.py", line 3, in <module>
    from . import activations
  File "C:\Users\mzm\Anaconda3\lib\site-packages\keras\activations.py", line 4, in <module>
    from . import backend as K
  File "C:\Users\mzm\Anaconda3\lib\site-packages\keras\backend\__init__.py", line 73, in <module>
    from .tensorflow_backend import *
  File "C:\Users\mzm\Anaconda3\lib\site-packages\keras\backend\tensorflow_backend.py", line 1, in <module>
    import tensorflow as tf
  File "C:\Users\mzm\Anaconda3\lib\site-packages\tensorflow\__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "C:\Users\mzm\Anaconda3\lib\site-packages\tensorflow\python\__init__.py", line 60, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "C:\Users\mzm\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    fp, pathname, description = imp.find_module('_pywrap_tensorflow', [dirname(__file__)])
  File "C:\Users\mzm\Anaconda3\lib\imp.py", line 296, in find_module
    raise ImportError(_ERR_MSG.format(name), name=name)
ImportError: No module named '_pywrap_tensorflow'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\mzm\Anaconda3\lib\site-packages\tensorflow\python\__init__.py", line 54, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Users\mzm\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 29, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\Users\mzm\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in swig_import_helper
    import _pywrap_tensorflow
ModuleNotFoundError: No module named '_pywrap_tensorflow'


Error importing tensorflow.  Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter
from there.


Does any know how to fix it??

**Comment by ShashwatHiregoudar** [comment]
Guys its simple,
Come out of the tensorflow environment
like - 
source deactivate tensorflow
then try your tensor board 
Ex:    tensorboard --logdir=/tmp/tboard/output

**Comment by yick2232** [comment]
https://github.com/ravivalluri/tensorflow/blob/master/tensorflow/g3doc/get_started/os_setup.md#import-error
There is a explain with this problem.

**Comment by Shiv751** [comment]
Traceback (most recent call last):
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 906, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: %1 is not a valid Win32 application.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\__init__.py", line 54, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow')
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\WorkspaceMaven\A-python\Test.py", line 1, in <module>
    import tensorflow as tf
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\__init__.py", line 60, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 906, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: %1 is not a valid Win32 application.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\__init__.py", line 54, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow')
  File "C:\Users\SEM\AppData\Local\Programs\Python\Python35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow'


Error importing tensorflow.  Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter
from there.

**Comment by newton3** [comment]
Hi, I got the same issue: 
```
Error importing tensorflow.  Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter
from there.
```

This was exactly after following the getting started guide and trying a import tensorflow as tf.
Wondering why is this issue closed when so many people still have the issue.
I couldn't find the solution anywhere on this thread or on stackoverflow.

Please help!!

**Comment by Shiv751** [comment]
Hi Abhishek,
I am still not resolved with this issue if you find potential fixes than
please let me know.

**Comment by Periter** [comment]
Oh,my solution is upgrading the pip to latest version， then use it to reinstall tensorflow

**Comment by cyberking18** [comment]
lol, you guys are just naive! It's clearly written that you cant import tensorflow on the terminal itself, C'mon go and create a python file and then compile it silly!

**Comment by LucianaMarques** [comment]
This thread seems somewhat old, but just to share the problem I had and how I solved it.

I first tried to import keras, but I faced the same error on this issue title. 

I uninstalled protobuf and updated tensorflow. It worked.

```
$ pip uninstall protobuf
$ pip install -U tensorflow
```

**Comment by Anusha-K11** [comment]
1.Download Python 3.5.2 (3.5.2 only!!)
2.In command prompt : pip install tensorflow 
(ignore this --Cache entry deserialization failed, entry ignored)
3.:python
4.>>import tensorflow as tf (
if no error then successfully installed
5.>>print(tf.__version__) 
(use double underscore above)
![Screenshot (205)](https://user-images.githubusercontent.com/48626909/59749961-9fcb5580-929b-11e9-88de-7debe7c02f0d.png)
)
